src/main/java/org/digitalstack/chess/ChessBoard.java

```diff
@@ -12,10 +12,28 @@ public ChessBoard() {
     }
 
     public void add(Pawn pawn, int xCoordinate, int yCoordinate, PieceColor pieceColor) {
-        throw new UnsupportedOperationException("Need to implement ChessBoard.add()");
+    	if(isLegalBoardPosition(xCoordinate, yCoordinate)) {
```

Comment:
Negate this condition and fail fast. You will remove one layer of imbrication this way.